src/ik.rs

```diff
@@ -208,8 +212,9 @@ where
         &self,
         arm: &SerialChain<T>,
         target_pose: &Isometry3<T>,
-        constraints_array: [bool; 6],
+        constraints: &Constraints,
     ) -> Result<DVector<T>, Error> {
+        let constraints_array = constraints_to_bool_array(constraints.clone());
```

Comment:
I do not want to clone() which has multiple joint names in the loop.
So you can change the function arguments of `constraints_to_bool_array` to get the reference instead of an instance.